### ISSUE TITLE
Datasource-grouping only works for first associated section

### ISSUE THREAD

**Issue by twiro** [symphonycms/selectbox_link_field#86]
### I have the following scenario:

- One "**Images**"-section is used to manage all images of two other sections "**Projects**" and "**News**".
- The entries in the "**Images**"-section are associated to a single entry of on of these two "parent"-sections. I use a (hidden) SBL-field called "entry" for that.
- I have a datasource called "images" that pulls all the images for the currently needed entries.
- On specific fronted pages I have to show images for multiple entries, so I want to group the entries in the "images"-datasource by their parent-entry.

### This results in the following buggy behavior:

- The image-entries associated to entries of the first parent-section "**Projects**" show up as expected in the datasource (grouped by parent-entry).
- The image-entries associated to entries of the second parent-section "**News**" simply disappear.

As soon as I turn off grouping all expected image-entries (of both parent-sections) are included in the datasource again. I also tried switching to a more fancy name than "entry" for the SBL-field - doesn't do a difference. I also tested this scenario with **Association Field** - obviously has the same bug.

### Setup:

- Symphony 2.6.11
- Selectbox Link Field 2.0.0
- Association Field 2.0.1

**Comment by nitriques** [comment]
Hum that's weird.
Do you have the same problem with Symphony 2.7.0 ?

**Comment by twiro** [comment]
> _Do you have the same problem with Symphony 2.7.0 ?_

Yep. Just replicated the exact same issue with the following Setup:

- Symphony 2.7.0
- Selectbox Link Field 2.0.1
- Association Field 2.0.2

**Comment by twiro** [comment]
A few screenshots to illustrate my test-setting:

### Section "Projects"

![screen shot 2017-08-17 at 22 09 07](https://user-images.githubusercontent.com/870227/29431993-36d19d76-839a-11e7-886b-870bb4e64fb4.png)

### Section "News"

![screen shot 2017-08-17 at 22 09 19](https://user-images.githubusercontent.com/870227/29432037-5211b904-839a-11e7-817d-36d8e246289d.png)

### Section "Images"

![screen shot 2017-08-17 at 22 05 39](https://user-images.githubusercontent.com/870227/29432052-61e2b1c6-839a-11e7-97d1-362d2d057e65.png)

### Section "Images" - "Parent Entry" Selectbox Link Field

![screen shot 2017-08-17 at 22 10 31](https://user-images.githubusercontent.com/870227/29432071-751b1eea-839a-11e7-836f-671e4f1d8bb3.png)

### Frontend Test Case

Now I build a datasource "projects" including all 4 entries from the "projects"-section, another datasource named "news" including all 4 entries from the "news"-section and a third datasource named "images" that is filtered by the ds-params of the first two datasources. The filter-settings look like this:

![screen shot 2017-08-17 at 22 06 16](https://user-images.githubusercontent.com/870227/29432147-ca3f32e4-839a-11e7-9e02-9975d8b2df85.png)

If a don't activate grouping the output of the datasource "images" looks like this:

![screen shot 2017-08-17 at 22 12 20](https://user-images.githubusercontent.com/870227/29432179-e89f30ea-839a-11e7-92c3-614bf664608e.png)

If i activate grouping by "parent entry" the output changes to this:

![screen shot 2017-08-17 at 22 11 58](https://user-images.githubusercontent.com/870227/29432197-fd96f834-839a-11e7-834c-bc5e558f644c.png)

So obviously datasource-grouping by entries of a SBL- or association-field doesn't work as expected if the  SBL- or association-field contains entries from multiple sections.

**Comment by nitriques** [comment]
> Doesn't work as expected if the SBL- or association-field contains entries from multiple sections.

The problem I see are missing records, but the order seems correct (A is before C). I am right ?

Also, what happens if you change the order direction (meaning it should find a News entry before a Project entry)

Thanks for those super awesomely detailed explanation @twiro
